qcodes/instrument_drivers/Keysight/Keysight_N5230A.py

```diff
@@ -0,0 +1,160 @@
+                           get_cmd=':SENS:FREQ:STAR?',
+                           set_cmd=':SENS:FREQ:STAR {}',
+                           get_parser=float,
+                           unit='Hz')
+
+        self.add_parameter('stop_frequency',
+                           label='Stop Frequency',
+                           get_cmd=':SENS:FREQ:STOP?',
+                           set_cmd=':SENS:FREQ:STOP {}',
+                           get_parser=float,
+                           unit='Hz')
+
+        self.add_parameter('span',
+                           label='Span',
+                           get_cmd=':SENS:FREQ:SPAN?',
+                           set_cmd=':SENS:FREQ:SPAN {}',
+                           get_parser=float,
+                           unit='Hz')
+
+        self.add_parameter('s_parameter',
+                           label='S parameter',
+                           docstring='Set the measured S parameter, allowed values are "S11", "S12", "S21", and "S22"',
+                           get_cmd=':CALC:PAR:CAT?',
+                           set_cmd=':CALC:PAR:MOD  {}',
+                           get_parser=lambda x: x[1:-1].split(',')[1])
+
+        self.add_parameter('num_points',
+                           label='Number of points',
+                           get_cmd=':SENS:SWE:POIN?',
+                           set_cmd=':SENS:SWE:POIN  {}',
+                           get_parser=int)
+
+        self.add_parameter('num_averages',
+                           label='Number of Averages of a sweep',
+                           get_cmd=':SENS1:AVER:COUN?',
+                           set_cmd=':SENS1:AVER:COUN {}',
+                           get_parser=int)
+
+        self.add_parameter('averaging',
+                           label='Enable Averaging',
+                           get_cmd=':SENS:AVER?',
+                           set_cmd=':SENS:AVER {}',
+                           val_mapping={'on': 1, 'off': 0})
+
+        self.add_parameter('start_sweep',
+                           label='Start Sweep',
+                           get_cmd='*STB?',
+                           set_cmd=(':SENS:SWE:TIME:AUTO ON;'
+                                    ':SENS:AVER:CLE;'
+                                    '*CLS;*ESE 1;'
+                                    ':SENS:SWE:GRO:COUN {};'
+                                    ':SENS:SWE:MODE GRO;'
+                                    '*OPC'),
+                           get_parser=lambda x: int(x) & 32 == 32
+                           )
+
+        self.connect_message()
+
+    def get_idn(self):
+        IDN = self.ask_raw('*IDN?')
+        vendor, model, serial, firmware = map(str.strip, IDN.split(','))
+        IDN = {'vendor': vendor, 'model': model,
+               'serial': serial, 'firmware': firmware}
+        return IDN
+
+    def _get_trace(self, verbose=False):
+
+        self.start_sweep(self.num_averages())
+        while not self.start_sweep():
```

Comment:
Does it make sense to make the interface clearer by implementing smth like `while not self.sweep_is_finished()` where `sweep_is_finished` is a method rather than a parameter, and `start_sweep` is also a method rather than a parameter?

Reply:
I agree , I don't think the get and set methods are a good match for each other. Such a complex set command also makes me slightly nervous. it seems very tailored to a specific use case. But I could be wrong